ipfs-9-to-10/migration/config_conv.go

```diff
@@ -0,0 +1,189 @@
+package mg9
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+
+	log "github.com/ipfs/fs-repo-migrations/stump"
+)
+
+var (
+	ip4BootstrapAddr  = "/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ"
+	quicBootstrapAddr = "/ip4/104.131.131.82/udp/4001/quic/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ"
+)
+
+// convArray does an inplace conversion of an array of strings
+// configuration from one version to another
+type convArray func([]string) []string
+
+// convertFile converts a config file from one version to another
+func convertFile(orig string, new string, enableQuic bool, convBootstrap convArray, convSwarm convArray) error {
+	in, err := os.Open(orig)
+	if err != nil {
+		return err
+	}
+	out, err := os.Create(new)
+	if err != nil {
+		return err
+	}
+	return convert(in, out, enableQuic, convBootstrap, convSwarm)
+}
+
+// convert converts the config from one version to another
+func convert(in io.Reader, out io.Writer, enableQuic bool, convBootstrap convArray, convSwarm convArray) error {
+	data, err := ioutil.ReadAll(in)
+	if err != nil {
+		return err
+	}
+	confMap := make(map[string]interface{})
+	if err = json.Unmarshal(data, &confMap); err != nil {
+		return err
+	}
+
+	// Convert bootstrap config
+	convertBootstrap(confMap, convBootstrap)
+
+	// Convert swarm config
+	if enableQuic {
+		// When enabling quic
+		// - Remove experimental option from config
+		confEnabled, ok := removeExperimentalQuic(confMap)
+		// - Only convert swarm config if experimental quic option was present
+		//   and not enabled
+		if ok && !confEnabled {
+			convertSwarm(confMap, convSwarm)
+		}
+	} else {
+		convertSwarm(confMap, convSwarm)
+	}
+
+	fixed, err := json.MarshalIndent(confMap, "", "  ")
+	if err != nil {
+		return err
+	}
+	out.Write(fixed)
+	out.Write([]byte("\n"))
+	return nil
+}
+
+// Remove Experimental.QUIC flag
+func removeExperimentalQuic(confMap map[string]interface{}) (bool, bool) {
+	confExpi := confMap["Experimental"].(map[string]interface{})
+	if confExpi == nil {
+		return false, false
+	}
+	enabledi, ok := confExpi["QUIC"]
+	enabled := enabledi.(bool)
```

Comment:
Will this crash if someone did  `"QUIC": "true"` (instead of `true`) ? Hopefully IPFS did not allow that mistake in the first place. Anyways, worth `enabled, ok :=` and checking rather than panicking.